exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -146,89 +158,121 @@
+  public static final OptionValidator IMPLICIT_SUFFIX_COLUMN_LABEL_VALIDATOR = new StringValidator(
+      IMPLICIT_SUFFIX_COLUMN_LABEL, "suffix");
+  public static final String IMPLICIT_FQN_COLUMN_LABEL = "drill.exec.storage.implicit.fqn.column.label";
+  public static final OptionValidator IMPLICIT_FQN_COLUMN_LABEL_VALIDATOR = new StringValidator(
+      IMPLICIT_FQN_COLUMN_LABEL, "fqn");
+  public static final String IMPLICIT_FILEPATH_COLUMN_LABEL = "drill.exec.storage.implicit.filepath.column.label";
+  public static final OptionValidator IMPLICIT_FILEPATH_COLUMN_LABEL_VALIDATOR = new StringValidator(
+      IMPLICIT_FILEPATH_COLUMN_LABEL, "filepath");
+
+  public static final String JSON_READ_NUMBERS_AS_DOUBLE = "store.json.read_numbers_as_double";
+  public static final BooleanValidator JSON_READ_NUMBERS_AS_DOUBLE_VALIDATOR = new BooleanValidator(
+      JSON_READ_NUMBERS_AS_DOUBLE, false,
+      "Reads numbers with or without a decimal point as DOUBLE. Prevents schema change errors.");
+
+  public static final String MONGO_ALL_TEXT_MODE = "store.mongo.all_text_mode";
+  public static final OptionValidator MONGO_READER_ALL_TEXT_MODE_VALIDATOR = new BooleanValidator(MONGO_ALL_TEXT_MODE,
+      false);
+  public static final String MONGO_READER_READ_NUMBERS_AS_DOUBLE = "store.mongo.read_numbers_as_double";
+  public static final OptionValidator MONGO_READER_READ_NUMBERS_AS_DOUBLE_VALIDATOR = new BooleanValidator(
+      MONGO_READER_READ_NUMBERS_AS_DOUBLE, false);
+  public static final String MONGO_BSON_RECORD_READER = "store.mongo.bson.record.reader";
+  public static final OptionValidator MONGO_BSON_RECORD_READER_VALIDATOR = new BooleanValidator(
+      MONGO_BSON_RECORD_READER, true);
+
+  public static final BooleanValidator ENABLE_UNION_TYPE = new BooleanValidator("exec.enable_union_type", false);
 
   // TODO: We need to add a feature that enables storage plugins to add their own options. Currently we have to declare
   // in core which is not right. Move this option and above two mongo plugin related options once we have the feature.
-  String HIVE_OPTIMIZE_SCAN_WITH_NATIVE_READERS = "store.hive.optimize_scan_with_native_readers";
-  OptionValidator HIVE_OPTIMIZE_SCAN_WITH_NATIVE_READERS_VALIDATOR =
+  public static final String HIVE_OPTIMIZE_SCAN_WITH_NATIVE_READERS = "store.hive.optimize_scan_with_native_readers";
+  public static final OptionValidator HIVE_OPTIMIZE_SCAN_WITH_NATIVE_READERS_VALIDATOR =
       new BooleanValidator(HIVE_OPTIMIZE_SCAN_WITH_NATIVE_READERS, false);
 
-  String SLICE_TARGET = "planner.slice_target";
-  long SLICE_TARGET_DEFAULT = 100000l;
-  PositiveLongValidator SLICE_TARGET_OPTION = new PositiveLongValidator(SLICE_TARGET, Long.MAX_VALUE,
-      SLICE_TARGET_DEFAULT);
+  public static final String SLICE_TARGET = "planner.slice_target";
+  public static final String SLICE_TARGET_DESCRIPTION =
+      "The number of records manipulated within a fragment before Drill parallelizes operations.";
```

Comment:
"The number of records processed by each fragment before Drill creates another parallel fragment."?